lib/simple_config.py

```diff
@@ -157,12 +157,12 @@ def convert_version_2(self):
         self.rename_config_keys(self.user_config, {'auto_cycle': 'auto_connect'})
 
         try:
-            # migrate server string FROM host:port:proto TO host:port
+            # change server string FROM host:port:proto TO host:port:s
             server_str = self.user_config.get('server')
             host, port, protocol = str(server_str).rsplit(':', 2)
             assert protocol in ('s', 't')
             int(port)  # Throw if cannot be converted to int
-            server_str = str('{}:{}'.format(host, port))
+            server_str = str('{}:{}:s'.format(host, port))
```

Comment:
I'm a little confused about this one.  Why would "convert_version_2" need to be rewritten to always hard-code ssl ports? Huh? Why this change.  Convince me it won't break anything...

Also some servers *are* non-ssl. Not clear on why the hardcoded 's' there.

Please elaborate if you can. Thanks.

Reply:
The string is actually required to have 3 fields, and `network.deserialize_server` will crash if it only has 2.  So the previous code was invalid.

Commit cba8052e7501803ebc49d458568dd96c14c7960b (cherry-picked from Electrum commit f77f029d728456e2a080abd7cf688afc270a20bc) was supposed to fix this, but somehow failed to include both files.

I don't know why those committers wanted to hard-code the 's', but it was clearly their intention.

Reply:
Right but the thing that gets me is:

- they parse the protocol; 's' or 't' 2 lines up.  
- 's' and 't' are still supported by extant code. 
- They basically throw away information and force 't' servers now to all be 's'.. why?

I'm going to ask ghost43 about this.  You know.. he could have just made a mistake. :)

Reply:
I don't understand it either, and yet the code, the comment, and the commit message all agree. Let's see what he says.